dev/run-tests

```diff
@@ -190,17 +202,21 @@ CURRENT_BLOCK=$BLOCK_SPARK_UNIT_TESTS
 
   echo "[info] Running Spark tests with these arguments: $SBT_MAVEN_PROFILES_ARGS ${SBT_MAVEN_TEST_ARGS[@]}"
 
-  # NOTE: echo "q" is needed because sbt on encountering a build file with failure
-  # (either resolution or compilation) prompts the user for input either q, r, etc
-  # to quit or retry. This echo is there to make it not block.
-  # NOTE: Do not quote $SBT_MAVEN_PROFILES_ARGS or else it will be interpreted as a
-  # single argument!
-  # "${SBT_MAVEN_TEST_ARGS[@]}" is cool because it's an array.
-  # QUESTION: Why doesn't 'yes "q"' work?
-  # QUESTION: Why doesn't 'grep -v -e "^\[info\] Resolving"' work?
-  echo -e "q\n" \
-    | build/sbt $SBT_MAVEN_PROFILES_ARGS "${SBT_MAVEN_TEST_ARGS[@]}" \
-    | grep -v -e "info.*Resolving" -e "warn.*Merging" -e "info.*Including"
+  if [ "${AMPLAB_JENKINS_BUILD_TOOL}" == "maven" ]; then
+    build/mvn test $SBT_MAVEN_PROFILES_ARGS --fail-at-end
```

Comment:
Actually it can't and this was the issue I alluded to in the main comments :/ Essentially the `SBT_MAVEN_TEST_ARGS` are an array of sub-projects to test through sbt although I couldn't find a way to get a maven build to leverage that same functionality. I tried various versions of the `-pl <subproject>` CLI options for maven, but they never correctly built all dependencies before executing and would only work properly if I ran a:

`mvn clean install`

first. I wasn't sure how Jenkins was maintaining state (i.e. does it remove the ~/.m2 folder after each build) so figured it was better to merely have a default "test all" setup in this instance.